pkg/clustermanager/eksa_mover.go

```diff
@@ -0,0 +1,165 @@
+package clustermanager
+
+import (
+	"context"
+	"math"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/clients/kubernetes"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/retrier"
+	"github.com/aws/eks-anywhere/pkg/types"
+)
+
+// MoverOpt allows to customize a Mover on construction.
+type MoverOpt func(*Mover)
+
+// Mover applies the cluster spec to the management cluster and waits
+// until the changes are fully reconciled.
+type Mover struct {
+	log                                        logr.Logger
+	clientFactory                              ClientFactory
+	applyClusterTimeout, waitForFailureMessage time.Duration
```

Comment:
do we need `waitForFailureMessage`?

also can we rename `applyClusterTimeout` to something more appropiate for this usecase? these names look like they make more sense in the applier